From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 8 Aug 2023 14:20:17 +0200
Subject: [PATCH 5/7] vconsole-setup: simplify error handling

no need to negate an error which we don't need negated.

also, add debug log about errors we ignore.
---
 src/vconsole/vconsole-setup.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/vconsole/vconsole-setup.c b/src/vconsole/vconsole-setup.c
index 911af61bb3..1c7af08b52 100644
--- a/src/vconsole/vconsole-setup.c
+++ b/src/vconsole/vconsole-setup.c
@@ -530,23 +530,23 @@ static int find_source_vc(char **ret_path, unsigned *ret_idx) {
 
                 r = verify_vc_allocation(i);
                 if (r < 0) {
-                        if (!err)
-                                err = -r;
+                        log_debug_errno(r, "VC %u existance check failed, skipping: %m", i);
+                        RET_GATHER(err, r);
                         continue;
                 }
 
                 sprintf(path, "/dev/tty%u", i);
                 fd = open_terminal(path, O_RDWR|O_CLOEXEC|O_NOCTTY);
                 if (fd < 0) {
-                        if (!err)
-                                err = -fd;
+                        log_debug_errno(fd, "Failed to open terminal %s, ignoring: %m", path);
+                        RET_GATHER(err, r);
                         continue;
                 }
 
                 r = verify_vc_kbmode(fd);
                 if (r < 0) {
-                        if (!err)
-                                err = -r;
+                        log_debug_errno(r, "Failed to check VC %s keyboard mode: %m", path);
+                        RET_GATHER(err, r);
                         continue;
                 }
 
